qiskit_nature/transformers/active_space_transformer.py

```diff
@@ -94,112 +129,152 @@ def transform(self, q_molecule: QMolecule) -> QMolecule:
+        # either freeze_core is specified
+        valid = self.freeze_core
+        # or at least num_electrons and num_molecular_orbitals must be valid
+        valid |= isinstance(self.num_electrons, int) and \
+            isinstance(self.num_molecular_orbitals, int)
+
+        return valid
+
+    def _extract_mo_occupation_vector(self, q_molecule: QMolecule):
         mo_occ_full = (q_molecule.mo_occ, q_molecule.mo_occ_b)
         if mo_occ_full[0] is None:
             # QMolecule provided by driver without `mo_occ` information available. Constructing
             # occupation numbers based on ground state HF case.
             occ_alpha = [1.] * q_molecule.num_alpha + [0.] * (q_molecule.num_orbitals -
                                                               q_molecule.num_alpha)
-            if beta:
+            if self._beta:
                 occ_beta = [1.] * q_molecule.num_beta + [0.] * (q_molecule.num_orbitals -
                                                                 q_molecule.num_beta)
             else:
                 occ_alpha[:q_molecule.num_beta] = [o + 1 for o in occ_alpha[:q_molecule.num_beta]]
                 occ_beta = None
             mo_occ_full = (np.asarray(occ_alpha), np.asarray(occ_beta))
-        mo_occ_total = mo_occ_full[0] + mo_occ_full[1] if beta else mo_occ_full[0]
+        return mo_occ_full
 
-        # compute number of inactive electrons
+    def _determine_active_space(self, q_molecule: QMolecule):
         nelec_total = q_molecule.num_alpha + q_molecule.num_beta
+
+        if self.freeze_core:
+            inactive_orbs_idxs = q_molecule.core_orbitals
+            if self.remove_orbitals is not None:
+                inactive_orbs_idxs.extend(self.remove_orbitals)
+            active_orbs_idxs = [o for o in range(q_molecule.num_orbitals)
+                                if o not in inactive_orbs_idxs]
+
+            # compute number of active electrons
+            nelec_inactive = sum([self._mo_occ_total[o] for o in inactive_orbs_idxs])
+            nelec_active = nelec_total - nelec_inactive
+
+            num_alpha = (nelec_active - (q_molecule.multiplicity - 1)) // 2
+            num_beta = nelec_active - num_alpha
+
+            self._num_particles = (num_alpha, num_beta)
+
+            return (active_orbs_idxs, inactive_orbs_idxs)
+
+        # compute number of inactive electrons
         nelec_inactive = nelec_total - self.num_electrons
         if self.num_alpha is not None:
-            if not beta:
+            if not self._beta:
                 warning = 'The provided instance of QMolecule does not provide any beta ' \
-                          + 'coefficients but you tried to specify a separate number of alpha ' \
-                          + 'electrons. Continuing as if it does not matter.'
+                          + 'coefficients but you tried to specify a separate number of alpha' \
+                          + ' electrons. Continuing as if it does not matter.'
```

Comment:
The reason for the comment above was that I was trying to understand this warning. Basically you are saying that if you gave alpha we were expecting an open shell molecule, but a closed shell computation has been done since there are not separate alpha and beta coeffs since they are the same (and hence no beta). Rather than saying `Continuing as if it does not matter` can we say something more like. 'The system has no orbitals that are single occupied by alpha electrons, the alpha electrons value supplied {self.num_alpha} will be be ignored`